### PR TITLE
Cleanup router a tiny little bit

### PR DESCRIPTION
**Pull request by luxzeitlos**
This is just a few types, replacing `for` loops with `for of` loops and moving some stuff from `init` to the constructor or property initializers.

### CODE REVIEW

File: packages/@ember/-internals/routing/lib/system/router.ts [hunk 1]
```diff
@@ -467,8 +493,8 @@ class EmberRouter extends EmberObject {
       'application',
       { path: '/', resetNamespace: true, overrideNameAssertion: true },
       function(this: EmberRouterDSL) {
-        for (let i = 0; i < dslCallbacks.length; i++) {
-          dslCallbacks[i].call(this);
+        for (let dslCallback of dslCallbacks) {
```
Comment (rwjblue) [inline packages/@ember/-internals/routing/lib/system/router.ts]: We cannot use `for...of` loops due to global `Symbol` requirement (`Symbol` is not present on IE11).

**Review by rwjblue — CHANGES_REQUESTED** [review]

File: packages/@ember/-internals/routing/lib/system/router.ts (hunk 1 above)
Comment (luxferresum) [reply packages/@ember/-internals/routing/lib/system/router.ts]: gnrgs I hate this world. Sorry.

File: packages/@ember/-internals/routing/lib/system/router.ts [hunk 2]
```diff
@@ -248,6 +248,11 @@ interface OutletState {
   outlets: NestedOutletState;
 }
 
+interface EngineInstance extends Owner {
```
Comment (rwjblue) [inline packages/@ember/-internals/routing/lib/system/router.ts]: technically I think its the other way around

**Review by rwjblue — APPROVED** [review]

**Comment by rwjblue** [comment]
thank you!

**Pull request merged**